contribs/gmf/src/directives/mobilebackgroundlayerselector.js

```diff
@@ -0,0 +1,131 @@
+goog.provide('gmf.MobileBackgroundLayerSelectorController');
+goog.provide('gmf.mobileBackgroundLayerSelectorDirective');
+
+goog.require('gmf');
+goog.require('gmf.Themes');
+goog.require('ngeo.BackgroundEventType');
+goog.require('ngeo.BackgroundLayerMgr');
+
+
+gmfModule.value('gmfMobileBackgroundLayerSelectorTemplateUrl',
+    /**
+     * @param {angular.JQLite} element Element.
+     * @param {angular.Attributes} attrs Attributes.
+     */
+    function(element, attrs) {
+      var templateUrl = attrs['gmfMobileBackgroundLayerSelectorTemplateurl'];
+      return templateUrl !== undefined ? templateUrl :
+          gmf.baseTemplateUrl + '/mobilebackgroundlayerselector.html';
+    });
+
+
+/**
+ * Provide a "mobile background layer selector" directive.
+ *
+ * @example
+ * <gmf-mobile-background-layer-selector
+ *   gmf-mobile-background-layer-selector-map="::ctrl.map">
+ * </gmf-mobile-background-layer-selector>
+ *
+ * @param {string} gmfMobileBackgroundLayerSelectorTemplateUrl Url to template.
+ * @return {angular.Directive} The Directive Definition Object.
+ * @ngInject
+ * @ngdoc directive
+ * @ngname gmfMobileBackgroundLayerSelector
+ */
+gmf.mobileBackgroundLayerSelectorDirective = function(
+    gmfMobileBackgroundLayerSelectorTemplateUrl) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      'getMobileMapFn': '&gmfMobileBackgroundLayerSelectorMap',
+      'getMobileBackgroundLayerSelectorOptionsFn':
+          '&gmfMobileBackgroundLayerSelectorOptions'
+    },
+    controller: 'GmfMobileBackgroundLayerSelectorController',
+    controllerAs: 'ctrl',
+    templateUrl: gmfMobileBackgroundLayerSelectorTemplateUrl
+  };
+};
+
+
+gmfModule.directive('gmfMobileBackgroundLayerSelector',
+    gmf.mobileBackgroundLayerSelectorDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {ngeo.BackgroundLayerMgr} ngeoBackgroundLayerMgr Background layer
+ *     manager.
+ * @param {gmf.Themes} gmfThemes Themes service.
+ * @export
+ * @ngInject
+ * @ngdoc controller
+ * @ngname GmfMobileBackgroundLayerSelectorController
+ */
+gmf.MobileBackgroundLayerSelectorController = function(
+    $scope, ngeoBackgroundLayerMgr, gmfThemes) {
+
+  var map = $scope['getMobileMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @private
+   */
+  this.map_ = map;
+
+  /**
+   * @type {ol.layer.Base}
+   */
+  this.bgLayer;
+
+  /**
+   * @type {Array.<ol.layer.Base>}
+   */
+  this.bgLayers;
+
+  /**
+   * @type {ngeo.BackgroundLayerMgr}
+   * @private
+   */
+  this.backgroundLayerMgr_ = ngeoBackgroundLayerMgr;
+  gmfThemes.getBgLayers().then(goog.bind(
+      /**
+       * @param {Array.<ol.layer.Base>} bgLayers Array of background
+       *     layer objects.
+       */
+      function(bgLayers) {
+        this['bgLayers'] = bgLayers;
```

Comment:
Seems to can get rid of the `[]` annotation for scope values with the `@export` annotation ?
See https://github.com/fgravin/ngeo/blob/guidelines/docs/guidelines.md#property-renaming

Can you give it a try ?